dags/ampliseq.py

```diff
@@ -1,103 +1,134 @@
-            writer.writerow(['study', 'timestamp', 'code_revision'])
-            for study in studies:
-                writer.writerow([study, current_timestamp, ampliseq_version])
+        @task
+        def update_provenance(studies):
+            fieldnames = ['study', 'timestamp', 'code_revision']
+            updated_studies = []
 
-        config_path = f"{base_path}ampliseq.config" # TODO validate exists
-        
-        commands = []
-        for i in range(len(studies)):
-            study = studies[i]
-            study_in_path = paths[i]
-            study_samplesheet_path = f"{study_in_path}/samplesheet.csv" # TODO validate exists
-            study_params_path = f"{study_in_path}/nf-params.json" # TODO validate exists
-            study_out_path = f"{study_in_path}/out"
-            study_work_dir = f"{study_in_path}/work"
+            # Read existing data
+            existing_data = []
+            if os.path.exists(PROVENANCE_PATH):
+                with open(PROVENANCE_PATH, 'r') as csvfile:
+                    reader = csv.DictReader(csvfile)
+                    existing_data = list(reader)
 
-            nextflow_command = (f"nextflow run nf-core/ampliseq -with-trace "
-                                f"-r {ampliseq_version} "
-                                f"-c {config_path} "
-                                f"-params-file {study_params_path} "
-                                f"-work-dir {study_work_dir} "
-                                f"--input {study_samplesheet_path} "
-                                f"--outdir {study_out_path} "
-                                f"-profile docker")
-            
-            # TODO dont want a hardcoded path here
-            R_command = (f"Rscript /data/MicrobiomeDB/mbio_airflow_dags/bin/ampliseq_postProcessing.R {study} {study_out_path}")
+            # Update or append new data
+            with open(PROVENANCE_PATH, 'w', newline='') as csvfile:
+                writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+                writer.writeheader()
+                for study in studies:
+                    updated_study = {
+                        'study': study['study'],
+                        'timestamp': study['current_timestamp'],
+                        'code_revision': AMPLISEQ_VERSION
+                    }
+                    updated_studies.append(updated_study)
+                
+                # Merge with existing data
+                all_data = {d['study']: d for d in existing_data}
+                all_data.update({d['study']: d for d in updated_studies})
+                
+                # Write back to the file
+                writer.writerows(all_data.values())
 
-            command = (nextflow_command + "; " + R_command)
+        loaded_studies = load_studies()
 
-            commands.append(command)
+        with TaskGroup("processing_tasks", tooltip="Processing tasks") as processing_tasks:  # Merged task groups
```

Comment:
i think we can improve the naming here as well.. what is the 'processing_task' doing? assume this dag grows one day and we might have multiple processing tasks. always favor overly specific and long names for things, as opposed to short or vague ones.

Reply:
some specific suggestions: the task group could be called 'process_amplicon_studies', the nextflow task within that group could be called 'run_ampliseq' and the rscript task could be called 'run_r_postprocessing'

Reply:
ill add another general rule w the naming of things.. putting 'task' in the name of a task isnt providing any further information to a reader (which might be your future self), so i wouldnt bother personally. you could try always starting names of tasks w a verb, if that helps.